s3torchconnector/tst/unit/test_s3_client.py

```diff
@@ -82,3 +84,34 @@ def test_user_agent_always_starts_with_package_version(comments):
     if comments_str:
         assert comments_str in s3_client.user_agent_prefix
         assert comments_str in s3_client._client.user_agent_prefix
+
+
+@given(
+    part_size=integers(min_value=5 * 1024, max_value=5 * 1024 * 1024),
+    throughput_target_gbps=floats(min_value=10.0, max_value=100.0),
+)
+def test_s3_client_custom_config(part_size: int, throughput_target_gbps: float):
+    # Part size must have values between 5MiB and 5GiB
+    part_size = part_size * 1024
```

Comment:
Not blocking: If it doesn't work to give a part size that's not a multiple of 1kb, we should add a test for that, otherwise we should do the multiplication in the given